x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/columns.tsx

```diff
@@ -6,195 +6,135 @@
-    description: i18n.DELETE_RULE,
-    icon: 'trash',
-    name: i18n.DELETE_RULE,
-    onClick: async (rowItem: Rule) => {
-      await deleteRulesAction([rowItem.id], dispatch, dispatchToaster);
-      await reFetchRules();
-      await refetchPrePackagedRulesStatus();
-    },
-  },
-];
-
-interface GetColumnsProps {
-  dispatch: React.Dispatch<RulesTableAction>;
-  formatUrl: FormatUrl;
-  hasMlPermissions: boolean;
+interface ColumnsProps {
   hasPermissions: boolean;
-  loadingRuleIds: string[];
-  navigateToApp: (appId: string, options?: NavigateToAppOptions | undefined) => Promise<void>;
-  hasReadActionsPrivileges: HasReadActionsPrivileges;
-  dispatchToaster: Dispatch<ActionToaster>;
-  reFetchRules: () => Promise<void>;
-  refetchPrePackagedRulesStatus: () => Promise<void>;
-  docLinks: DocLinksStart;
 }
 
-const getColumnEnabled = ({
-  hasMlPermissions,
-  hasReadActionsPrivileges,
-  dispatch,
-  hasPermissions,
-  loadingRuleIds,
-}: GetColumnsProps): TableColumn => ({
-  field: 'enabled',
-  name: i18n.COLUMN_ACTIVATE,
-  render: (_, rule: Rule) => (
-    <EuiToolTip
-      position="top"
-      content={getToolTipContent(rule, hasMlPermissions, hasReadActionsPrivileges)}
-    >
-      <RuleSwitch
-        data-test-subj="enabled"
-        dispatch={dispatch}
-        id={rule.id}
-        enabled={rule.enabled}
-        isDisabled={
-          !canEditRuleWithActions(rule, hasReadActionsPrivileges) ||
-          !hasPermissions ||
-          (isMlRule(rule.type) && !hasMlPermissions && !rule.enabled)
-        }
-        isLoading={loadingRuleIds.includes(rule.id)}
-      />
-    </EuiToolTip>
-  ),
-  width: '95px',
-  sortable: true,
-});
+const useEnabledColumn = ({ hasPermissions }: ColumnsProps): TableColumn => {
+  const hasMlPermissions = useHasMlPermissions();
+  const hasActionsPrivileges = useHasActionsPrivileges();
+  const rulesTableContext = useRulesTableContext();
+  invariant(
+    rulesTableContext,
+    'useRulesTableContext should be used inside RulesTableContextProvider'
+  );
 
-const getColumnRuleName = ({ navigateToApp, formatUrl }: GetColumnsProps): TableColumn => ({
-  field: 'name',
-  name: i18n.COLUMN_RULE,
-  render: (value: Rule['name'], item: Rule) => (
-    <EuiToolTip content={value} anchorClassName="eui-textTruncate">
-      <LinkAnchor
-        data-test-subj="ruleName"
-        onClick={(ev: { preventDefault: () => void }) => {
-          ev.preventDefault();
-          navigateToApp(APP_UI_ID, {
-            deepLinkId: SecurityPageName.rules,
-            path: getRuleDetailsUrl(item.id),
-          });
-        }}
-        href={formatUrl(getRuleDetailsUrl(item.id))}
-      >
-        {value}
-      </LinkAnchor>
-    </EuiToolTip>
-  ),
-  width: '38%',
-  sortable: true,
-  truncateText: true,
-});
+  const {
+    state: { loadingRulesAction, loadingRuleIds },
+  } = rulesTableContext;
+
+  const loadingIds = useMemo(
+    () =>
+      loadingRulesAction != null &&
```

Comment:
nit: looks like you don't need this condition

Reply:
Right, removed it